### PR TITLE
Google Auth

### PR DESCRIPTION
**Pull request by naisargi17**
### Description
Google authentication 

### Changes Made
firebase is use for google authentication 

### Checklist 

- [ X] I have tested the changes locally
- [ X] Documentation has been updated (if necessary)
- [ X] Changes are backward-compatible

### Additional Notes

Footer

### CODE REVIEW

**Comment by VaibhavArora314** [comment]
resolve merge conflicts and following errors
image

**Comment by naisargi17** [comment]
yes , I 'm working on but I facing issues here also.
image

**Comment by naisargi17** [comment]
Hi @VaibhavArora314  can assist me where is this confilct coming from ?

**Comment by VaibhavArora314** [comment]
> Hi @VaibhavArora314 can assist me where is this confilct coming from ?

there is an error in usePosts hook. also resolve merge conflicts.
image

**Review by VaibhavArora314 — CHANGES_REQUESTED** [review]
dont pass your firebase credentials, pass them as env variables
remove package-lock.json from root of the project
resolve merge conflicts

File: frontend/src/firebase.tsx
Comment (VaibhavArora314) [inline frontend/src/firebase.tsx:1]: the secrets here should not be directly written in code, put them as environment variables

**Review by VaibhavArora314 — CHANGES_REQUESTED** [review]
also, resolve merge conflicts

**Review by Ultimateutkarsh11 — APPROVED** [review]
Kindly resolve the merge conflicts, rest is fine.

File: frontend/src/firebase.tsx
Comment (naisargi17) [reply frontend/src/firebase.tsx:1]: Yes I'm working on it ...there is some updates in firebase so I'm working on it .

**Comment by naisargi17** [comment]
> Kindly resolve the merge conflicts, rest is fine.

I'm resolving it .please give me some time .

**Pull request closed without merging**